lib/location/location.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <assert.h>
+#include <logging/log.h>
+#include <modem/location.h>
+
+#include "loc_core.h"
+
+LOG_MODULE_REGISTER(location, CONFIG_LOCATION_LOG_LEVEL);
+
+/** @brief Semaphore protecting the use of location requests. */
+K_SEM_DEFINE(loc_core_sem, 1, 1);
+
+int location_init(location_event_handler_t handler)
```

Comment:
Do we need `location_deinit()`? NRF has plenty of examples which has both init/deinit but not sure if there are those with only one init.
And should this be register instead of init?

Reply:
at_notif does only have init()

Reply:
We don't need deinit.